girder/utility/filesystem_assetstore_adapter.py

```diff
@@ -70,6 +70,20 @@ def validateInfo(doc):
             raise ValidationException(
                 'Unable to write into directory "%s".' % doc['root'])
 
+        if doc['perms'] is None:
```

Comment:
For supporting legacy assetstores, we can't assume the existence of this key, so we should change this to:

    if doc.get('perms') is None: